client/src/app/plan.service.ts

```diff
@@ -99,6 +99,57 @@ export class PlanService {
     return this.user.pipe(map(mapCallback));
   }
 
+  /**
+   * Adds a new semester into years chronologically. If term already exists, nothing is done.
+   * @param term The term in string form
+   * @param yearIndex  the Index of the year that the semester will be added to
+   * @param years The years array
+   */
+  addNewSemester(term: string, yearIndex: number, years: Array<any>): void {
+    const semesterDoesNotAlreadyExist = years[yearIndex].semesters.every(
+      (semester) => {
+        return semester.term.toLowerCase() !== term.toLowerCase();
+      }
+    );
+
+    if (semesterDoesNotAlreadyExist) {
+      const newSemester = {
+        term: term.toLowerCase(),
+        year:
+          this.TERMS[term.toLowerCase()] > 2
+            ? years[yearIndex].beginning + 1
+            : years[yearIndex].beginning,
+        difficulty: 0,
+        units: 0,
+        status: "planned",
+        courses: [],
+      };
+
+      years[yearIndex].semesters.push(newSemester);
+
+      this.sortSemestersChronologically(yearIndex, years);
+    } else {
+      window.alert("Semester already exists!");
+    }
+  }
+
+  /**
+   * Chronologically sorts the semesters at the given index of years
+   * @param yearIndex The index of the year in the year array
+   * @param years The years array
+   */
+  private sortSemestersChronologically(
```

Comment:
Helper function that will be reused by other functions. It sorts the semesters chronologically. (Summer, Fall, Winter, Spring)